src/socket.rs

```diff
@@ -109,6 +114,38 @@ impl Socket {
         self.url.clone()
     }
 
+    pub fn is_connected(&self) -> bool {
+        self.status() == Status::Connected
+    }
+
+    pub fn is_waiting_to_reconnect(&self) -> bool {
+        self.status() == Status::WaitingToReconnect
+    }
+
+    pub fn has_never_connected(&self) -> bool {
+        self.status() == Status::NeverConnected
+    }
+
+    pub fn is_disconnected(&self) -> bool {
+        self.status() == Status::Disconnected
+    }
+
+    pub fn is_shutting_down(&self) -> bool {
+        self.status() == Status::ShuttingDown
+    }
+
+    pub fn is_shutdown(&self) -> bool {
+        self.status() == Status::ShutDown
+    }
+
+    pub fn status(&self) -> Status {
+        self.status.get()
+    }
+
+    pub fn statuses(&self) -> broadcast::Receiver<Result<Status, Arc<tungstenite::Error>>> {
```

Comment:
Arc<Result<Status, tungstenite::Error> is generally preferable over wrapping error types in Arc

Reply:
I have `Arc<tungstenite::Error>` because the `tungstenite::Error` also needs to be sent in the `ConnectError`, so making it an `Arc` allow me to return it from `socket_connect`:

https://github.com/liveview-native/phoenix-channels-client/blob/e45ec143f991120955480706e591fa5cb6397b21/src/socket/listener.rs#L696-L700

which is sent back through the `connect_tx` 
https://github.com/liveview-native/phoenix-channels-client/blob/e45ec143f991120955480706e591fa5cb6397b21/src/socket/listener.rs#L181-L200

to the tell the `socket.connect` caller why the initial connect failed

https://github.com/liveview-native/phoenix-channels-client/blob/e45ec143f991120955480706e591fa5cb6397b21/src/socket.rs#L156-L169

https://github.com/liveview-native/phoenix-channels-client/blob/e45ec143f991120955480706e591fa5cb6397b21/src/socket.rs#L323-L327

If you can suggest a way to get the `tungenstenite::Error` that *isn't* `Clone` to both the `ConnectError::WebSocketError` and the `Socket::statuses()` subscribers I'd be open to that suggestion, but this change to make the whole `Result` and `Arc` won't allow the `tungstenite::Error` to be shared between those two channels.